src/Geta.Optimizely.ProductFeed/Configuration/IEndpointRouteBuilderExtensions.cs

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) Geta Digital. All rights reserved.
+// Licensed under Apache-2.0. See the LICENSE file in the project root for more information
+
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Routing;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Geta.Optimizely.ProductFeed.Configuration
+{
+    public static class IEndpointRouteBuilderExtensions
+    {
+        public static IEndpointRouteBuilder MapProductFeeds(this IEndpointRouteBuilder endpoints)
+        {
+            var descriptors = endpoints.ServiceProvider.GetServices<FeedDescriptor>();
+
+            foreach (var descriptor in descriptors)
+            {
+                endpoints.MapControllerRoute(
+                    descriptor.Name + "-endpoint",
+                    descriptor.FileName.TrimStart('/'),
```

Comment:
How URLs will look like?

Reply:
at the end url of the feed could look like `https://localhost/googleproductfeed`.
trimming everywhere is added so that we have proper server name (without ending trail slash) and user can also customize feed name as wished: `/googleproductfeed` or `googleproductfeed`.